seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java

```diff
@@ -221,6 +227,15 @@ public Object computeForValue(Expression expression, Object[] inputFields) {
             Parenthesis parenthesis = (Parenthesis) expression;
             return computeForValue(parenthesis.getExpression(), inputFields);
         }
+        if (zetaSQLFilter.isConditionExpr(expression)) {
+            return zetaSQLFilter.executeFilter(expression, inputFields);
+        }
```

Comment:
Why add this? Is we missed something?

Reply:
`case when id = 1 then 0 else 1`
circulation Iterate to the syntax id=1 filter，I have optimized the code writing